gapic/samplegen/samplegen.py

```diff
@@ -86,48 +92,147 @@ class InvalidRequestSetup(SampleError):
+           assignment definitions. The only difference in the bottommost dicts
+           are that "field" maps only to the second part of a dotted variable name.
+           Other key/value combinations in the dict are unmodified for the time being.
+
+           TODO: Conduct module lookup and expansion for protobuf enums.
+                 Requires proto/method/message descriptors.
+           TODO: Permit single level field/oneof requst parameters.
+                 Requires proto/method/message descriptors.
+           TODO: Add/transform to list repeated element fields.
+                 Requires proto/method/message descriptors.
+
+           E.g. [{"field": "clam.shell", "value": "10 kg", "input_parameter": "shell"},
+                 {"field": "clam.pearls", "value": "3"},
+                 {"field": "squid.mantle", "value": "100 kg"}]
+                  ->
+                [TransformedRequest("clam",
+                  [{"field": "shell", "value": "10 kg", "input_parameter": "shell"},
+                   {"field": "pearls", "value": "3"}]),
+                 TransformedRequest("squid", [{"field": "mantle", "value": "100 kg"}])]
+
+           The transformation makes it easier to set up request parameters in jinja
+           because it doesn't have to engage in prefix detection, validation,
+           or aggregation logic.
+
+
+        Args:
+            request (list[dict{str:str}]): The request body from the sample config
+
+        Returns:
+            list[dict{str:(str|list[dict{str:str}])}]: The transformed request block.
+
+        Raises:
+            RedefinedVariable: If an "input_parameter" attempts to redefine a
+                               previously defined variable.
+            ReservedVariableName: If an "input_parameter" value or a "base" value
+                                  is a reserved word.
+            InvalidRequestSetup: If a dict in the request lacks a "field" key
+                                 or the corresponding value is malformed.
+        """
+        base_param_to_attrs: Mapping[str,
+                                     List[Mapping[str, str]]] = defaultdict(list)
+        for field_assignment in request:
+            field_assignment_copy = dict(field_assignment)
+            input_param = field_assignment_copy.get("input_parameter")
+            if input_param:
+                self._handle_lvalue(input_param)
+
+            field = field_assignment_copy.get("field")
+            if not field:
+                raise InvalidRequestSetup(
+                    "No field attribute found in request setup assignment: {}",
+                    field_assignment_copy)
+
+            # TODO: properly handle top level fields
+            # E.g.
+            #
+            #  -field: edition
+            #   comment: The edition of the series.
+            #   value: '123'
+            #   input_parameter: edition
+            m = re.match(r"^([^.]+)\.([^.]+)$", field)
```

Comment:
Eventually you'll need to that last group to be repeated to allow for subfields, indexing, etc. Fine to defer it. Great if you mention this in the TODO.

Reply:
Will do. There's a lot of complexity that will be added once the code starts examining the message types.